tests/workflow_contracts/runner_policy_test.py

```
@@ -0,0 +1,394 @@
+
+    checkout = _steps(job)[0]
+    assert str(checkout.get("uses", "")).startswith("actions/checkout@")
+    assert checkout.get("with") == {"fetch-depth": 0}, (
+        "the check diffs against the base ref, so it needs full history"
+    )
+
+    script = _steps(job)[1].get("run")
+    assert isinstance(script, str), "the check must declare a script"
+    assert "git diff" in script and "git log" in script, (
+        "the check must remain a git-history inspection, not a build"
+    )
```

Ruff (0.16.1)

[warning] Assertion should be broken down into multiple parts

(PT018)